ql/src/java/org/apache/hadoop/hive/ql/io/AcidUtils.java

```diff
@@ -2839,6 +2848,87 @@ public static void setNonTransactional(Map<String, String> tblProps) {
     tblProps.remove(hive_metastoreConstants.TABLE_TRANSACTIONAL_PROPERTIES);
   }
 
+  /**
+   * Look for delta directories matching the list of writeIds and deletes them.
+   * @param rootPartition root partition to look for the delta directories
+   * @param conf configuration
+   * @param writeIds list of writeIds to look for in the delta directories
+   * @return list of deleted directories.
+   * @throws IOException
+   */
+  public static List<FileStatus> deleteDeltaDirectories(Path rootPartition, Configuration conf, Set<Long> writeIds)
+      throws IOException {
+    FileSystem fs = rootPartition.getFileSystem(conf);
+
+    PathFilter filter = (p) -> {
+      String name = p.getName();
+      for (Long wId : writeIds) {
+        if (name.startsWith(deltaSubdir(wId, wId)) && !name.contains("=")) {
```

Comment:
Why the contains "=", are we checking for a partition where the user named the column exactly like a valid delta dir? I don't think we should support that

Reply:
I was also wondering the same, as this code was there in the earlier patches so I have just kept it. We can remove this.

Reply:
changed, also excluded base directory from listing